assets/src/components/resizable-box/index.js

```diff
@@ -34,13 +47,64 @@ export default ( props ) => {
 				bottom: true,
 				left: false,
 			} }
-			onResizeStop={ ( event, direction, elt, delta ) => {
+			onResizeStop={ ( event, direction ) => {
+				const { deltaW, deltaH } = getResizedWidthAndHeight( event, angle, lastSeenX, lastSeenY, direction );
 				onResizeStop( {
-					width: parseInt( width + delta.width, 10 ),
-					height: parseInt( height + delta.height, 10 ),
+					width: parseInt( width + deltaW, 10 ),
+					height: parseInt( height + deltaH, 10 ),
+					positionTop: parseInt( blockElement.style.top, 10 ),
+					positionLeft: parseInt( blockElement.style.left, 10 ),
 				} );
 			} }
-			onResizeStart={ onResizeStart }
+			onResizeStart={ ( event, direction, element ) => {
+				lastSeenX = event.clientX;
+				lastSeenY = event.clientY;
+				blockElement = element.closest( '.wp-block' );
+				blockElementTop = blockElement.style.top;
+				blockElementLeft = blockElement.style.left;
+				onResizeStart();
+			} }
+			onResize={ ( event, direction, element ) => {
+				const { deltaW, deltaH } = getResizedWidthAndHeight( event, angle, lastSeenX, lastSeenY, direction );
+
+				const appliedWidth = minWidth <= width + deltaW ? width + deltaW : minWidth;
+				const appliedHeight = minHeight <= height + deltaH ? height + deltaH : minHeight;
+
+				if ( angle ) {
+					const radianAngle = getRadianFromDeg( angle );
+
+					// Compare position between the initial and after resizing.
+					const initialPosition = getBlockPositioning( width, height, radianAngle );
+					const resizedPosition = getBlockPositioning( appliedWidth, appliedHeight, radianAngle );
+					const diff = {
+						left: resizedPosition.left - initialPosition.left,
+						top: resizedPosition.top - initialPosition.top,
+					};
+					// Get new position based on the difference.
+					const originalPos = {
+						left: getPixelsFromPercentage( 'x', parseInt( blockElementLeft, 10 ) ),
+						top: getPixelsFromPercentage( 'y', parseInt( blockElementTop, 10 ) ),
+					};
+
+					// @todo Figure out why calculating the new top / left position doesn't work in case of small height value.
+					// @todo Remove this temporary fix.
```

Comment:
@swissspidy I changed the max and min sizes for resizing based on the values in your PR #2296. Also added this check here for when height is less than 60, since that's when the shift starts, however, I'm not sure why it happens. Added this fix here for now since might not have time to figure out the reason behind it. Let me know if it seems to work a bit better now.